policies/coding-style.md

```diff
@@ -591,21 +591,89 @@ should be used. This is more commonly referred to as C90. ISO/IEC 9899:1999
 (also known as C99) is not supported on some platforms that OpenSSL is
 used on and therefore should be avoided.
 
-## Chapter 15: Miscellaneous
+## Chapter 15: Expressions
 
-Do not use `!` to check if a pointer is NULL, or to see if a str...cmp
-function found a match.  For example, these are wrong:
+Avoid needless parentheses.<br/>
+For example, do not write
+```c
+    if ((p == NULL) && (!f(((2 * x) + y) == (z++))))
+ ```
+but
+```c
+    if (p == NULL && !f(2 * x + y == z++)).
+```
+
+For clarity, always put parentheses when mixing `&&` and `||` operations.<br/>
+For example,
+ ```c
+    if ((a && b) || c)
+ ```
+
+Put parentheses around macro bodies that can be used as expressions.<br/>
+Put parentheses around uses of macro arguments
+(unless they are passed on as-is to a further macro or function).<br/>
+For example,
+ ```c
+#define SQUARE(a) ((a) * (a))
+```
+
+In comparisons with constants (including `NULL`)
+place the constant on the right-hand side of the comparison operator.<br/>
+For example,
+```c
+    while (i++ < 10 && p != NULL)
+```
 
+Do not use implicit checks for
+numbers (not) being `0` or pointers (not) being `NULL`.<br/>
+For example, do not write
+```c
+    if (i)
+    if (!(x & MASK))
+    if (!strcmp(a, "FOO"))
+    if (!(p = BN_new()))
+```
+but do this instead:
 ```c
-    if (!(p = BN_new())) ...
-    if (!strcmp(a, "FOO")) ...
+    if (i != 0)
+    if ((x & MASK) == 0)
+    if (strcmp(a, "FOO") == 0)
+    if ((p = BN_new()) == NULL)
 ```
 
-Do this instead:
+If you need to break an expression into multiple lines,
+make the line break before an operator, not after.
+It is preferred that such a line break is made
+before as low priority an operator as possible.<br>
+Examples:
 
+* not this:
+  ```c
+  if (somewhat_long_function_name(foo) == 1 && a_long_variable_name
+      == 2)
+  ```
+  but rather:
+  ```c
+  if (somewhat_long_function_name(foo) == 1
+      && a_long_variable_name == 2)
+  ```
+
+* This is, however, still ok:
+  ```c
+  if (this_thing->this_freakishly_super_long_name(somewhat_long_name, 3)
+      == PRETTY_DARN_LONG_MACRO_NAME)
+  ```
+
+Whan appearing at the beginning of a line,
+the binary logical operators `&&` and `||`
+may get an extra indentation (+4 characters).<br/>
```

Comment:
Is there a reason why those operators are singled out?

Reply:
It's generalized in the last commit:

````Markdown
When appearing at the beginning of a line,
operators may get an extra indentation (+4 characters).
For example,
```c
    if (long_condition_expression_1
            && condition_expression_2) {
        statement_1;
        statement_2;
    }
```
````

I think this comment can be resolved.

Reply:
Yup